src/components/metadata/styles/metadata-code.scss

```diff
@@ -0,0 +1,184 @@
+@import '../../../styles/variables';
+@import '../../../styles/extends';
+@import '../../../styles/mixins';
+@import './metadata-code-themes.scss';
+
+.kui-theme--light {
+  --color-metadata-code-bg: #{$color-bg-light-4};
+}
+
+.kui-theme--dark {
+  --color-metadata-code-bg: #{$color-bg-dark-3};
+}
+
+.pipeline-metadata-code {
+  @extend %sidebar;
+
+  $margin: 24px;
+  position: absolute;
+  top: -1px; /* Avoids pixel rounding gaps */
+  right: 0;
+  bottom: -1px;
+
+  // On small screens anchor to the left side of the screen
+  left: 0;
+
+  z-index: 2;
+  padding: 0;
+  background: var(--color-metadata-code-bg);
+  transform: translateX(100vw);
+  transition: transform ease 0.5s 0.1s, left ease 0.5s;
+
+  &--visible {
+    transform: translateX(0);
+  }
+
+  @mixin breakpoints($sidebar-width) {
+    // When screen is the minimum usable width
+    @media (min-width: $metadata-sidebar-code-breakpoint) {
+      // Anchor code to the left sidebar
+      left: $sidebar-width + $margin;
+    }
+
+    // When screen is wide enough to fit all sidebars
+    $sidebar-total-width: $sidebar-width + $margin + $metadata-sidebar-code-open +
+      $metadata-sidebar-width-open;
+    @media (min-width: #{$sidebar-total-width}) {
+      // Anchor to right sidebar
+      left: calc(
+        100% - #{$margin + $metadata-sidebar-code-open +
+          $metadata-sidebar-width-open}
+      );
+    }
+  }
+
+  // When sidebar is open
+  &--sidebarVisible {
+    @include breakpoints($sidebar-width-open);
+  }
+
+  // When sidebar is closed
+  &--no-sidebarVisible {
+    @include breakpoints($sidebar-width-closed);
+  }
+
+  @media print {
+    display: none;
+  }
+}
+
+.pipeline-metadata-code__title {
+  flex-grow: 1;
+  margin: 0;
+  margin: 2.1em 36px 1.8em 36px;
```

Comment:
This should align with using the same unit (`em` ) - @richardwestenra ?

Reply:
Why do you say that? I think this makes sense to use em for the vertical margin (which is based on the relative font-size), but use px for the horizontal margin (which is based on the container's padding, which is based on the overall container width, which is handled in pixels).

Reply:
Yeah exactly it's a bit nuanced I know but my approach to spacing usually depends on context. I tend to use `em` for vertical spacing as that's naturally related to typography and vertical rhythm. But really it's not a big deal in practice anymore, but I still find it's still a good way of expressing the intent.

Reply:
Thanks for the clarification @richardwestenra - I am quite confused as to our current preference of unit within the codebase - are we using px or em? My understanding is that it is either em or px as a consistent unit within the codebase, and if we are making exceptions for context then it's very easy for us to fall into the trap of individual interpretation?

Reply:
That's fair - I've added this to the list of things to add to the styleguide in KED-2413.

- We typically use `em` for font-sizes, and sizing things that might want to be set relative to the font-size - e.g. text margin-top/bottom, line-height, etc. This is partly because it is more accessible, and partly because it allows us to set the font-size in one place (i.e. on the `kedro` class) and then if we want to change later then everything else will update based on that value.
- We typically use `px` and `%` for layout - i.e. things that are more dependant on the size of the viewport than on the size of the font.

Reply:
I see, these 2 types of cases makes sense. In this case can we generally categorize to use `px` / `%`  for all layout-related fields ( i.e fields that would stay constant regardless of font sizing), and `em` for all font related / font-affected values?